### PR TITLE
Development

### PR DESCRIPTION
**Pull request by pieterDekker**
finished iteration 2

### CODE REVIEW

**Comment by DanChirtoaca** [comment]
This Pull Request does not follow the Official Git Workflow. Take a look at section 4 on release branches: [https://public.3.basecamp.com/p/FMvbSscgKCdjNtreihgwbCAz](url)

File: sampleExtension/statistics/tracker.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            self.getCurrentTab();
+            if(self.compareUrlToRegex(self.zeeguuRegex, self.tabActive)){
+                TrackerStorage.incrementDayStat(1);
+            }
+        }
+    };
+
+    // Function attached to the idle-listener. Sets the self.idle variable.
+    this.checkIdle = function(idleState){
+        if(idleState == 'active'){
```
Comment (lauraBaakman) [inline sampleExtension/statistics/tracker.js]: What if idleState is `idlestate`?

File: sampleExtension/statistics/trackerStorage.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    };
+
+
+    // Loads dummy data in the previous day list for demonstration purposes.
+    this.loadDummyData = function(amount){
+        var dummyList = [];
+        for(var i = amount; i > 0; i--){
+            var date = new Date();
+            date = new Date(date.setDate(date.getDate()-i));
+            dummyList.push({'date': ((date.getDate())+"/"+(date.getMonth()+1)+"/"+date.getFullYear()), 'timespent': Math.floor((Math.random()*100)+1)});
```
Comment (lauraBaakman) [inline sampleExtension/statistics/trackerStorage.js]: The value associated with the key `date` is incomprehensible, move to a function.

File: sampleExtension/statistics/trackerStorage.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    // Increments the counter for time spent on exercises today with 'amount'.
+    // When there is no currentDateStatistic in the storage yet, create one.
+    // And when the currentDateStatistic does not correspond with the actual current date,
+    // this currentDataStatistic is added to the list of previous day statistics.
+    this.incrementDayStat = function(amount){
+        self.getCurrentDayStatistic().then(function(response){
+            var today = self.getToday();
+            var responseObject = response.tds_currentDayStatistic;
+            if(responseObject == null){
```
Comment (lauraBaakman) [inline sampleExtension/statistics/trackerStorage.js]: To many levels of abstraction.

File: sampleExtension/statistics/trackerStorage.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            newList.push(dayStats);
+            self.setDayStatisticsList(newList);
+        });
+    };
+
+
+    // Get the list containing information about time spent on exercises. For the previous days, and the current day.
+    this.getCompleteDayStatList = function(){
+        return new Promise(function(resolve, reject){
+            self.getDayStatisticsList().then(function(response){
```
Comment (lauraBaakman) [inline sampleExtension/statistics/trackerStorage.js]: Whut?

File: sampleExtension/statistics/trackerStorage.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                var previousDays = response.tds_dayStatistics;
+                self.getCurrentDayStatistic().then(function(response2){
```
Comment (lauraBaakman) [inline sampleExtension/statistics/trackerStorage.js]: `response2` is not an acceptable variable name.

File: sampleExtension/statisticsPage/statistics.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+// Log console messages to the background page console instead of the content page.
+var console = chrome.extension.getBackgroundPage().console;
+
+var bg = chrome.extension.getBackgroundPage();
+
+//Local variables that holds the list of links and interceptCounter.
+var links = [];
+var interceptionCounter = 0;
+var interceptDateList = [];
+var date24 = 0;
```
Comment (lauraBaakman) [inline sampleExtension/statisticsPage/statistics.js]: What is the difference between `date24`, `date7` and `date31`?

File: sampleExtension/statisticsPage/statistics.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    chrome.tabs.query({active: true, currentWindow: true}, function (arrayOfTabs) {
+        // since only one tab should be active and in the current window at once
+        // the return variable should only have one entry
+        var activeTab = arrayOfTabs[0];
+        var activeTabUrl = activeTab.url;
+        bg.addToBlockedSites(activeTabUrl);
+    });
+}
+
+generateHtmlTableRow = function(site) {
```
Comment (lauraBaakman) [inline sampleExtension/statisticsPage/statistics.js]: Good attempt at isolating the generation of table rows, but avoid html in your javascript.

File: sampleExtension/statisticsPage/statistics.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+};
+
+calcInterceptData = function() {
+    var tmp = interceptDateList;
+    if (tmp != null) {
+        var oneDay = 24 * 60 * 60 * 1000; // hours*minutes*seconds*milliseconds
+        var firstDate = new Date();
+        var length = tmp.length;
+
+        for (var i = 0; i < length; i++) {
```
Comment (lauraBaakman) [inline sampleExtension/statisticsPage/statistics.js]: Too much is hapenning here.

**Comment by Mantsje** [comment]
We've had a look at all the comments and implemented a lot of changes. This pull-request does not adhere to the standards required by the git-workflow so we are closing this pull request. Please have a look at the new one

**Pull request closed without merging**